synapseclient/multipart_upload.py

```diff
@@ -329,18 +348,34 @@ def _multipart_upload(syn, filename, contentType, get_chunk_function, md5, fileS
     time_upload_started = time.time()
     progress=True
     retries=0
-    mp = Pool(8)
+
+    try:
+        mp = Pool(8)
+        expiredValue = Value(c_bool, False)
+        singleThreaded = False
+    except OSError:
+        mp = DummyPool()
+        expiredValue = DummyValue(False)
+        singleThreaded = True
     try:
         while retries<MAX_RETRIES:
             ## keep track of the number of bytes uploaded so far
-            completed = Value('d', min(completedParts * partSize, fileSize))
 
-            printTransferProgress(completed.value, fileSize, prefix='Uploading', postfix=filename)
-            chunk_upload = lambda part: _upload_chunk(part, completed=completed, status=status, 
+            if singleThreaded:
```

Comment:
this if else block should no longer be necessary after the refactor changes suggested above.